### PR TITLE
feat: add home page

### LINKED ISSUE #55: feat(home menu): home page
### What do you want to see?

That one page that has links to donate, create a group, donate, and the likes.

image

### PR DESCRIPTION
**Pull request by Rexogamer**
## Please make sure to check the following tasks before opening and submitting a PR

* [x] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [x] I have tested my changes locally and they are working as intended
* [x] These changes do not have any notable side effects on other Revolt projects

This pull request fixes #55.

To-do:
* [ ] some styling fixes for the CategoryButtons (text contrast on hover, fixed width) - are these out of scope?
* [x] code cleanup
* [ ] a bit of functionality (feedback/settings pages) - probably out of scope for this PR

### CODE REVIEW

File: components/client/index.tsx [hunk 1]
```diff
@@ -37,3 +37,6 @@ export const IS_REVOLT =
   // future proofing
   import.meta.env.VITE_API_URL === "https://app.revolt.chat/api" ||
   import.meta.env.VITE_API_URL === "https://revolt.chat/api";
+
+// TODO: should this use an env variable?
```
Comment (insertish) [inline components/client/index.tsx]: Use `import.meta.env.DEV`

File: packages/client/src/interface/Home.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                            <Match when={showLoungeButton && !isInLounge}>
+                                <CategoryButton
+                                    description={t("app.home.join-testers_desc")}
+                                    icon={<BiSolidRightArrowCircle size={24} />}>
+                                    {t("app.home.join-testers")}
+                                </CategoryButton>
+                            </Match>
+                        </Switch>
+                        <CategoryButton
+                            onClick={() => window.location.href = "https://insrt.uk/donate?utm_source=revolt_client_v2"} // TODO: probably want to use a different functions for this
```
Comment (insertish) [inline packages/client/src/interface/Home.tsx]: Should open in new tab / window, also `utm_source=revoltapp` is sufficient.

File: packages/client/src/interface/Home.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            <Content>
+                <Typography
+                    style={"text-align: center; margin-bottom: 1em; font-size: 200%;"}
+                    variant="legacy-settings-title">
+                    {t("app.special.modals.onboarding.welcome")}
+                    <br />
+                    <img style="margin-top: 0.5em;" height={80} src="/assets/wide.svg" />
+                </Typography>
+                <Buttons>
+                    <Column style={"margin-right: 0.25em;"}>
```
Comment (insertish) [inline packages/client/src/interface/Home.tsx]: Hard-coded styles throughout code, if really necessary, just wrap the component:
```
const Abc = styled(Column)` margin-right: 0.25em; `;
```
Also, use margin-inline-end or whatever the equivalent is.

File: packages/client/src/interface/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    return (
+        <Base>
+            <Header palette="primary">Home</Header>
+            <Content>
+                <Typography
+                    style={"text-align: center; margin-bottom: 1em; font-size: 200%;"}
+                    variant="legacy-settings-title">
+                    {t("app.special.modals.onboarding.welcome")}
+                    <br />
+                    <img style="margin-top: 0.5em;" height={80} src="/assets/wide.svg" />
```
Comment (insertish) [inline packages/client/src/interface/Home.tsx]: Also hard-coded styles, just pull them out like:
```
const Image = styled.img` margin-block-start: 0.5em; height: 80px; `;
```

**Review by insertish — CHANGES_REQUESTED** [review]

**Comment by Rexogamer** [comment]
Marking as ready for review - see the PR description

**Review by insertish — APPROVED** [review]

**Pull request merged**